ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/dns_settings.schema.yml

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2023 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  dns_settings:
+    documentation_options:
+      table: management-settings
+    type: dict
+    description: |-
+      DNS settings
+      For DNS source-interfaces see "source_interfaces.domain_lookup"
+    keys:
+      servers:
+        type: list
+        $ref: eos_cli_config_gen#/keys/ip_name_servers
+        items:
+          type: dict
+          keys:
+            vrf:
+              type: str
+              description: |-
+                VRF Name.
+                Can be used in combination with "use_mgmt_interface_vrf" and "use_inband_mgmt_vrf" to configure the DNS server under multiple VRFs.
+              convert_types:
+                - int
```

Comment:
VRF names can be numeric. So we always convert from int.

Reply:
what about making it a list of extra vrfs ?

Reply:
I doubt it will ever be used. And if someone really needs it, they can just add the server twice with different VRFs.

Reply:
fair enough. Though if I add the server twice, I could end up with duplicate VRF then given that the set is applied only at the `server` level (not saying we guard against it in SNMP!):

```
    dns_settings:
      domain: <str>
      servers:
        - vrf: MGMT
           ip_address: 1.1.1.1
        - use_mgmt_interface_vrf: true
           ip_address: 1.1.1.1
```

but I guess we can live with it.

Reply:
Ah I think the duplicate detection is much more important than having multiple vrfs on a single server. So maybe we should add primary_key on the list of servers using the ip_address. It would rule out adding two extra VRFs, but we always have structured config.